core/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodeFiltersTests.java

```diff
@@ -115,4 +114,56 @@ public void starMatch() {
         DiscoveryNode node = new DiscoveryNode("name1", "id1", DummyTransportAddress.INSTANCE, ImmutableMap.<String, String>of(), Version.CURRENT);
         assertThat(filters.match(node), equalTo(true));
     }
+
+    @Test
+    public void ipFilteringMatchingAnd() {
+        Settings settings = Settings.settingsBuilder()
+                .put("xxx.tag", "A")
+                .put("xxx._ip", "127.0.0.1")
+                .build();
+        DiscoveryNodeFilters filters = DiscoveryNodeFilters.buildFromSettings(AND, "xxx.", settings);
+
+        DiscoveryNode node = new DiscoveryNode("name1", "id1", DummyTransportAddress.INSTANCE,
+                ImmutableMap.of("tag", "A"), Version.CURRENT);
+        assertThat(filters.match(node), equalTo(true));
+    }
+
+    @Test
+    public void ipFilteringNotMatchingAnd() {
+        Settings settings = Settings.settingsBuilder()
+                .put("xxx.tag", "A")
+                .put("xxx._ip", "8.8.8.8")
+                .build();
+        DiscoveryNodeFilters filters = DiscoveryNodeFilters.buildFromSettings(AND, "xxx.", settings);
+
+        DiscoveryNode node = new DiscoveryNode("name1", "id1", DummyTransportAddress.INSTANCE,
+                ImmutableMap.of("tag", "A"), Version.CURRENT);
+        assertThat(filters.match(node), equalTo(false));
+    }
+
+    @Test
+    public void ipFilteringMatchingOr() {
+        Settings settings = Settings.settingsBuilder()
+                .put("xxx._ip", "127.0.0.1")
+                .put("xxx.tag", "A")
+                .build();
+        DiscoveryNodeFilters filters = DiscoveryNodeFilters.buildFromSettings(OR, "xxx.", settings);
+
+        DiscoveryNode node = new DiscoveryNode("name1", "id1", DummyTransportAddress.INSTANCE,
```

Comment:
I think this relies on NetworkUtils#getLocalHostName returning 127.0.0.1, but I don't think it's guaranteed ...